gensim/models/nmf.py

```diff
@@ -0,0 +1,643 @@
+
+        Returns
+        -------
+        list of (int, float)
+            The relevant topics represented as pairs of their ID and their assigned probability, sorted
+            by relevance to the given word.
+
+        """
+        if minimum_probability is None:
+            minimum_probability = self.minimum_probability
+        minimum_probability = max(minimum_probability, 1e-8)
+
+        # if user enters word instead of id in vocab, change to get id
+        if isinstance(word_id, str):
+            word_id = self.id2word.doc2bow([word_id])[0][0]
+
+        values = []
+
+        word_topics = self._W.getrow(word_id)
+
+        if self.normalize and word_topics.sum() > 0:
+            word_topics /= word_topics.sum()
+
+        for topic_id in range(0, self.num_topics):
+            word_coef = word_topics[0, topic_id]
+
+            if word_coef >= minimum_probability:
+                values.append((topic_id, word_coef))
+
+        return values
+
+    def get_document_topics(self, bow, minimum_probability=None):
+        """Get the topic distribution for the given document.
+
+        Parameters
+        ----------
+        bow : list of (int, float)
+            The document in BOW format.
+        minimum_probability : float
+            Topics with an assigned probability lower than this threshold will be discarded.
+
+        Returns
+        -------
+        list of (int, float)
+            Topic distribution for the whole document. Each element in the list is a pair of a topic's id, and
+            the probability that was assigned to it.
+
+        """
+        if minimum_probability is None:
+            minimum_probability = self.minimum_probability
+        minimum_probability = max(minimum_probability, 1e-8)
+
+        # if the input vector is a corpus, return a transformed corpus
+        is_corpus, corpus = utils.is_corpus(bow)
+
+        if is_corpus:
+            kwargs = dict(minimum_probability=minimum_probability)
+            return self._apply(corpus, **kwargs)
+
+        v = matutils.corpus2csc([bow], len(self.id2word)).tocsr()
+        h, _ = self._solveproj(v, self._W, v_max=np.inf)
+
+        if self.normalize:
+            h.data /= h.sum()
+
+        return [
+            (idx, proba.toarray()[0, 0])
+            for idx, proba in enumerate(h[:, 0])
+            if not minimum_probability or proba.toarray()[0, 0] > minimum_probability
+        ]
+
+    def _setup(self, corpus):
+        """Infer info from the first document and initialize matrices.
+
+        Parameters
+        ----------
+        corpus : iterable of list(int, float)
+            Training corpus.
+
+        """
+        self._h, self._r = None, None
+        first_doc_it = itertools.tee(corpus, 1)
+        first_doc = next(first_doc_it[0])
+        first_doc = matutils.corpus2csc([first_doc], len(self.id2word))[:, 0]
+        self.n_features = first_doc.shape[0]
```

Comment:
wait, why?

Reply:
still a question @anotherbugmaster, can you clarify please

Reply:
Didn't get it, what is the question?

Reply:
I can use id2word to infer number of tokens in a dictionary if that'll make it clear.

Reply:
yes, that's be better (btw - how you infer it from doc, it's in BoW format, or I did miss something?)

Reply:
Refactored a bit, please review agagin